VRDR/DeathRecord.cs

```diff
@@ -2956,6 +2956,92 @@ public string DateOfBirth
             }
         }
 
+        /// <summary>Decedent's Date of Birth Date Part Absent Extension.</summary>
+        /// <value>the decedent's date of birth date part absent reason</value>
+        /// <example>
+        /// <para>// Setter:</para>
+        /// <para>ExampleDeathRecord.DateOfBirthDatePartReason = "1940-02-19";</para>
+        /// <para>// Getter:</para>
+        /// <para>Console.WriteLine($"Decedent Date of Birth Date Part Reason: {ExampleDeathRecord.DateOfBirthDatePartAbsent}");</para>
+        /// </example>
+        [Property("Date Of Birth Date Part Absent", Property.Types.TupleArr, "Decedent Demographics", "Decedent's Date of Birth Date Part.", true, "http://build.fhir.org/ig/HL7/vrdr/StructureDefinition-VRDR-Decedent.html", true, 14)]
+        [FHIRPath("Bundle.entry.resource.where($this is Patient)._birthDate.extension.where(url='http://hl7.org/fhir/us/vrdr/StructureDefinition/VRDR-Partial-date-part-absent-reason')", "_birthDate")]
+        public Tuple<string,string>[] DateOfBirthDatePartAbsent
+        {
+            get
+            {     
+                if (Decedent.BirthDateElement != null){
+                    Extension datePartAbsent = Decedent.BirthDateElement.Extension.Where(ext => ext.Url == "http://hl7.org/fhir/us/vrdr/StructureDefinition/VRDR-Partial-date-part-absent-reason").FirstOrDefault();
+                    List<Tuple<string, string>> dateParts = new List<Tuple<string, string>>();
+                    if (datePartAbsent != null)
+                    {
+                        Extension yearAbsentPart = datePartAbsent.Extension.Where(ext => ext.Url == "year-absent-reason").FirstOrDefault();
+                        Extension monthAbsentPart = datePartAbsent.Extension.Where(ext => ext.Url == "month-absent-reason").FirstOrDefault();
+                        Extension dayAbsentPart = datePartAbsent.Extension.Where(ext => ext.Url == "day-absent-reason").FirstOrDefault();
+                        Extension yearPart = datePartAbsent.Extension.Where(ext => ext.Url == "date-year").FirstOrDefault();
+                        Extension monthPart = datePartAbsent.Extension.Where(ext => ext.Url == "date-month").FirstOrDefault();
+                        Extension dayPart = datePartAbsent.Extension.Where(ext => ext.Url == "date-day").FirstOrDefault();
+                        // Year part
+                        if (yearAbsentPart != null)
+                        {
+                            dateParts.Add(Tuple.Create("year-absent-reason", yearAbsentPart.Value.ToString())); // TODO change this to the code system value?
```

Comment:
Ultimately these will be coded values, right?   Where do you do that conversion?

Reply:
Do you mean the conversion to IJE? The BirthDate_Part_Absent_Get function in IJEMortality checks for a date absent part and will return a string of "9"s the length of the given IJE field if there is an absent reason.